### DISCUSSION TITLE
combining answers into one api value

### DISCUSSION THREAD

**Discussion by airplanenoise** [kennym/cf7-to-api#37] [Ideas]
hi, 

i am going to try it myself but maybe we can change the fields to the right of the API connexions to be a textarea, so that we can build an array in there? we have an API element that's actually an array. i don't really see a way to pre-process the answers and compile them into one shortcode...

thanks!

Kaz

**Comment by kennym** [comment]
I don't understand what you mean... can you give us a code example?

**Reply by airplanenoise** [reply]
thanks for the quick reply. 

here's an example - we have a form that needs to be sent to an API. one of the API fields is a questionnaire, and the questionnaire excepts a JSON array, like this : 

` [questionnaire] => [   {      "que":"Enter Your Question Here",      "ans":"Enter Your Answer"   },   {      "que":"Enter Your Question Here",      "ans":"Enter Your Answer"   } ]
`
the goal would be to build this array from various questions in the contact form, like this : 

` [questionnaire] => [   {      "que":"[question-1-label]",      "ans":"[question-1-answer]"   },   {      "que":"[question-2-label]",      "ans":"[question-2-answer]"   } ]
`
where the label is the actual question value, and the answer is what the person replied... 

seems very doable. 

i am looking into the plugin code and am on line 272 of the class-cf7-api-admin.php to try to replace input by a textarea, we could then write our own json code in that area. 

thoughts?

**Reply by airplanenoise** [reply]
something like this to start, 

```                <tr>
                            <th style="text-align:left;"><?php echo $mail_tag->name;?></th>
                            <td><?php if($mail_tag->name == 'questionnaire'):?>
									<textarea id="sf-<?php echo $mail_tag->name;?>" name="qs_wpcf7_api_map[<?php echo $mail_tag->name;?>]" class="large-text"><?php echo isset($wpcf7_api_data_map[$mail_tag->name]) ? $wpcf7_api_data_map[$mail_tag->name] : "";?></textarea>
								<?php else: ?>
									<input type="text" id="sf-<?php echo $mail_tag->name;?>" name="qs_wpcf7_api_map[<?php echo $mail_tag->name;?>]" class="large-text" value="<?php echo isset($wpcf7_api_data_map[$mail_tag->name]) ? $wpcf7_api_data_map[$mail_tag->name] : "";?>" />
								<?php endif; ?>
							</td>
                </tr>
```
based on the mail tag (a bit hacky). 

how would i them be able to aggregate several form answers into one api field? 

:) 

Kaz

**Reply by airplanenoise** [reply]
any luck @kennym ? :)
